typescript/infra/src/govern/ProxiedRouterGovernor.ts

```diff
@@ -54,19 +56,29 @@ export class ProxiedRouterGovernor<
   }
 
   protected handleEnrolledRouterViolation(violation: RouterViolation) {
-    const remoteDomain = this.checker.multiProvider.getDomainId(
-      violation.remoteChain,
-    );
+    const domains: number[] = [];
+    const addresses: string[] = [];
+
+    for (const [remoteChain, expectedRouter] of Object.entries(
+      violation.routerDiff,
+    ) as [ChainName, Address][]) {
+      const remoteDomain = this.checker.multiProvider.getDomainId(remoteChain);
+      domains.push(remoteDomain);
+      addresses.push(expectedRouter);
+    }
+
     return {
       chain: violation.chain,
       call: {
         to: violation.contract.address,
         data: violation.contract.interface.encodeFunctionData(
-          'enrollRemoteRouter',
-          [remoteDomain, violation.expected],
+          'enrollRemoteRouters',
+          [domains, addresses],
         ),
         value: BigNumber.from(0),
-        description: `Enroll router for remote chain ${violation.remoteChain} (${remoteDomain}) ${violation.expected} in ${violation.contract.address}`,
+        description: `Enroll missing routers for ${
```

Comment:
nit: iiuc this code path is also intended for when the routers change for whatever reason, not necessarily that routers are missing. I think this wording may be misleading for that situation, and it may be nice to surface the old router (though feel less strongly there, as we didn't do this before)

Reply:
fixed